tests/btb/recommendation/test_recommender.py

```diff
@@ -31,14 +31,19 @@ def setUp(self):
 
     @patch('btb.recommendation.recommender.NMF')
     def test___init__(self, nmf_mock):
+        index = 1
         nmf_mock().fit_transform.return_value = self.dpp_ranked
         # Run
-        recommender = Recommender(self.dpp_matrix, self.n_components)
+        with patch('numpy.random.randint', return_value=index):
```

Comment:
I think that it's better to follow the decorator approach, to validate the call arguments and to clearly separate set-up from running and assertion phases.
Also, it's better to limit the patch to the tested module. This means patching `btb.recommendation.recommender.np.random` rather than `numpy.random.randint`.

All together would be:
```
@patch('btb.recommendation.recommender.np.random'.randint)
@patch('btb.recommendation.recommender.NMF')
def test___init__(self, nmf_mock, randint_mock):
    # set-up
    index = 1
    nmf_mock().fit_transform.return_value = self.dpp_ranked
    randint_mock.return_value = index

    # run
    recommender = Recommender(self.dpp_matrix, self.n_components)

    # asserts
    randint_mock.assert_called_once_with(4)    # where 4 is self.dpp_matrix length
    np.testing.assert_array_equal(recommender.matching_dataset, self.dpp_matrix[index, :])

    # rest of the asserts...

Reply:
@csala Ok, I've changed it.